deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java

```diff
@@ -75,27 +84,77 @@ public void test_duration_to_header_value() {
 
     @ParameterizedTest
     @CsvSource({
+        "0, 0",
+        "0.0, 0",
         "1, 1000000000",
+        "01, 1000000000",
+        "09, 9000000000",
+        "1.0, 1000000000",
+        "1.00000, 1000000000",
+        " 123.4567890246, 123456789024",
         "' 2. ', 2000000000",
         "3.0, 3000000000",
         "3.1, 3100000000",
         "1234567890.12345, 1234567890123450112",
         "1.523, 1523000000",
         "'   1.523  ', 1523000000",
+        "1234567890123467890123467890123467890123467890123467890123467890, 9223372036854775807",
+        "12345678901234678901234678901234678901234678901234678901234678901.123467890123467890, 9223372036854775807",
```

Comment:
we're implicitly capping max at `Long.MAX_VALUE`, may want to document edge cases

Reply:
we could probably add something to the public javadoc on [Deadlines.parseFromRequest](https://github.com/palantir/deadlines-java/blob/develop/deadlines/src/main/java/com/palantir/deadlines/Deadlines.java#L120-L135) (that actually has no mention of what the type of the header value is intended to be, which is perhaps an oversight)